app/main.py

```diff
@@ -1,8 +1,36 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+        self.name = name
+        self.age = age
+
+        self.people[self.name] = self
+
+
+def find_partner(person_obj: Person,
+                 person_to_find: str | None
+                 ) -> Person | str | None:
+    if person_to_find and person_to_find in person_obj.people.keys():
+        partner = person_obj.people[person_to_find]
+        if hasattr(partner, "wife"):
+            partner.wife = person_obj
+        elif hasattr(partner, "husband"):
+            partner.husband = person_obj
+        return person_obj.people[person_to_find]
```

Comment:
When accessing a class attribute , it is generally considered best practice to use the class name instead of an instance (self) to access that attribute.

Reply:
thank you for answering question "why", i didn't know it